### PR TITLE
Add libevent 2.1.12

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **libevent/2.1.12**

2.1.12 does not change significantly from 2.1.11. It fixes a few patched problems so 2 of 3 patches are excluded.
Also, I consolidated versions to the `all` directory. Differences between versions are handled via patches. Initially, I used `2.1.11` instead of `all` hoping about packaging 2.0 but it is old and superseded by 2.1.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libevent/all/patches/0004-cmake-fix-runtime-install.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+diff --git a/cmake/AddEventLibrary.cmake b/cmake/AddEventLibrary.cmake
+index 04f5837e..88a19ca9 100644
+--- a/cmake/AddEventLibrary.cmake
++++ b/cmake/AddEventLibrary.cmake
+@@ -42,7 +42,7 @@ macro(export_install_target TYPE LIB_NAME OUTER_INCLUDES)
+         install(TARGETS "${LIB_NAME}_${TYPE}"
+             LIBRARY DESTINATION "lib" COMPONENT lib
+             ARCHIVE DESTINATION "lib" COMPONENT lib
+-            RUNTIME DESTINATION "lib" COMPONENT lib
++            RUNTIME DESTINATION "bin" COMPONENT lib
```
Comment (madebr) [inline recipes/libevent/all/patches/0004-cmake-fix-runtime-install.patch:10]: Is this fixing the location of the main dll or of some helper dll's?
Can this patch be sent upstream?

Comment (theirix) [reply recipes/libevent/all/patches/0004-cmake-fix-runtime-install.patch:10]: Don't really know. It was introduced by your commit 71ef2e30b096a44bb8ef811de4b314ca6d8648dd . I think all DLLs should be packaged in one directory, not the only one.

Comment (madebr) [reply recipes/libevent/all/patches/0004-cmake-fix-runtime-install.patch:10]: Haha, I think I was fixing the location of the main dll. I don't know of any other library.

Comment (SSE4) [reply recipes/libevent/all/patches/0004-cmake-fix-runtime-install.patch:10]: on other hand, I think we can leave everything as is, and just set `cpp_info.libdirs` and `cpp_info.bindirs`

**Pull request merged**